### ISSUE TITLE
How to substitute an instrument in a song?

### ISSUE THREAD

**Issue by oddtime** [hydrogen-music/hydrogen#1073]
Sometimes user may want to substitute an instrument with another, keeping the notes.
The way I would do it is adding the new instrument, select and moving all the notes to it and deleting the old one.
Is there another way?

**Comment by oddtime** [comment]
I originally tried pressing some key (ctrl, alt...) while dropping a new instrument, or seeing if there was a "change instrument" option in right click menu,
but the first model you describe seems faster to use to me

> show the gaps between instruments differently from the instruments
> highlight the gap or instrument that the mouse cursor is currently over

maybe just highlight the gap (as a thick line) or the instrument when mouse is over is sufficient?

> but it's not intuitively obvious what should happen to any notes associated with the dragged instrument. Discarded? Merged?

Why discarded? I would keep the notes or I don't understand

**Comment by theGreatWhiteShark** [comment]
Yes! Especially if have a full song at hand and want to test e.g. a different hihat or a whole different drumkit the lack of this feature is very annoying. I solve it using a [elisp function](https://github.com/theGreatWhiteShark/configurations-and-scripts/blob/master/emacs/elisp/hydrogen.el).

Regarding the second approach: With the mapping in #1070 one could reload the drumkit in order to swap/replace instruments.

But in general I would also favor the first approach and keep/assign the notes to the new instrument. The only problem I do see here is that from a user perspective I would expect to be able to drag instruments from any drumkit into the pattern editor and not just by the one loaded.

**Comment by oddtime** [comment]
Thanks for the elisp function

> The only problem I do see here is that from a user perspective I would expect to be able to drag instruments from any drumkit into the pattern editor and not just by the one loaded.

Yes, instruments should be dragged from the drumkit library panel on the right,
Instead dragging instrument from the pattern window should just change the order like now

**Comment by oddtime** [comment]
I am linking another issue #168 that asks the same feature so we can close both together
